stevensStringLib.h

```diff
@@ -1148,8 +1148,39 @@ namespace stevensStringLib
         return std::equal(str.begin(), str.begin() + str.size()/2, str.rbegin());
     }
 
-
-    //replace
+    /**
+    * Replaces characters using a vector syntax. Use multi_type_vector to construct proper input.
+    * Example syntax:
+    * 
+    * std::string testing = stevensStringLib::replace_characters({"testing {0}{1}", "1", "This is another test"}); // replaces string selection with specified characters
+    * std::cout << testing << std::endl;
+    * 
+    * @param vector<str> - example_vector above
+    * @retval str - returns replaced string
+    */
+    std::string replace_characters( std::vector<std::string> values ) 
+    {
+        std::string result = values[0];  // setting intital input
+        values.erase(values.begin()); // removes input value from vector
+
+        size_t index = 0;
+        size_t startPos = 0;
+        while ((startPos = result.find('{', startPos)) != std::string::npos) {
```

Comment:
For the first version of this function I think using curly braces here is fine, but I may add additional parameters to this function for people who want to change how they're forming their format strings.